semver.py

```diff
@@ -174,6 +174,39 @@ def __iter__(self):
         for v in self._astuple():
             yield v
 
+    def __getitem__(self, index):
+        """Implement getitem. If the part requested is undefined, or a part of the
+           range requested is undefined, it will throw an index error.
+           Negative indices are not supported
+
+           :param Union[int, slice] index: a positive integer indicating the
+                  offset or a slice
+           :raises: IndexError, if index is beyond the range or a part is None
+           """
```

Comment:
This docstring has an unusual indentation. According to [PEP 257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings) indentation may look like:
```python
def foo():
    """First line of comment

    Second line of comment. The longer description.
    """
```
(what I like less but seems to be favored by Guido & friends), or:
```python
def foo():
    """
    First line of comment

    Second line of comment. The longer description.
    """
```

Same below, in the method following next.